src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java

```diff
@@ -71,9 +70,18 @@ public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
             .transformToUniAndConcatenate(
                 readDocument -> deleteDocument(queryExecutor, delete, readDocument))
             .collect()
-            .asList();
-    return ids.onItem()
-        .transform(deletedIds -> new DeleteOperationPage(deletedIds, boolenFlag.get()));
+            .in(
+                AtomicInteger::new,
+                (atomicCounter, flag) -> {
+                  if (flag) {
+                    atomicCounter.incrementAndGet();
+                  }
+                });
+
+    return counter
+        .onItem()
+        .transform(
+            deletedCounter -> new DeleteOperationPage(deletedCounter.get(), boolenFlag.get()));
```

Comment:
Tiny typo: `boolenFlag` -> `booleanFlag` (was already there)

Reply:
renamed as moreData, in line with response message